pages/api/activities/index.js

```diff
@@ -23,7 +25,17 @@ export default async function handler(request, response) {
   if (request.method === "POST") {
     try {
       if (session) {
+        const existingUser = await User.findOne({ userId });
+        const user = existingUser
+          ? existingUser
+          : await User.create({
+              userId: userId,
+              name: userName,
+              picture: picture,
+            });
```

Comment:
this can be written a bit more concise: 
```suggestion
        const user = existingUser ?? await User.create({
              userId: userId,
              name: userName,
              picture: picture,
            });
```
The ?? operator or "nullish coalescence" operator returns the first value as long its not null / undefined. Otherwise it returns the second value.